scene/3d/light_3d.cpp

```diff
@@ -191,6 +199,25 @@ Light3D::BakeMode Light3D::get_bake_mode() const {
 void Light3D::set_projector(const Ref<Texture2D> &p_texture) {
 	projector = p_texture;
 	RID tex_id = projector.is_valid() ? projector->get_rid() : RID();
+
+#ifdef DEBUG_ENABLED
+	if (Object::cast_to<AnimatedTexture>(*p_texture)) {
+		WARN_PRINT(vformat("AnimatedTexture is not supported as a Light3D projector texture (%s).", get_path()));
+	} else if (Object::cast_to<AtlasTexture>(*p_texture)) {
+		WARN_PRINT(vformat("AtlasTexture is not supported as a Light3D projector texture (%s).", get_path()));
+	} else if (Object::cast_to<CameraTexture>(*p_texture)) {
+		WARN_PRINT(vformat("CameraTexture is not supported as a Light3D projector texture (%s).", get_path()));
+	} else if (Object::cast_to<CanvasTexture>(*p_texture)) {
+		WARN_PRINT(vformat("CanvasTexture is not supported as a Light3D projector texture (%s).", get_path()));
+	} else if (Object::cast_to<MeshTexture>(*p_texture)) {
+		WARN_PRINT(vformat("MeshTexture is not supported as a Light3D projector texture (%s).", get_path()));
+	} else if (Object::cast_to<Texture2DRD>(*p_texture)) {
+		WARN_PRINT(vformat("Texture2DRD is not supported as a Light3D projector texture (%s).", get_path()));
+	} else if (Object::cast_to<ViewportTexture>(*p_texture)) {
+		WARN_PRINT(vformat("ViewportTexture is not supported as a Light3D projector texture (%s). As a workaround, use `get_image()` on the ViewportTexture to assign the returned value to the Light3D projector property, but keep in mind this is a slow operation.", get_path()));
```

Comment:
This is the kind of thing I would expect to see on a Note in the class reference.
I don't think it's out of place here, just a bit of a mouthful. And that's not mentioning how long it gets when the node's path is included.
```suggestion
		WARN_PRINT(vformat("ViewportTexture cannot be used as a Light3D projector texture (%s). As a workaround, create an ImageTexture from the ViewportTexture's `get_image()`.", get_path()));
```
The "slow operation" part is important too, of course, it's just... a lot of information to cram into a warning.